docs/content/preview/contribute/core-database/build-from-src-centos.md

```diff
@@ -51,62 +51,125 @@ Update packages on your system, install development tools and additional package
 ```sh
 sudo yum update -y
 sudo yum groupinstall -y 'Development Tools'
-sudo yum install -y ruby perl-Digest epel-release ccache git python2-pip python-devel python3 python3-pip python3-devel which
-sudo yum install -y cmake3 ctest3 ninja-build
+sudo yum install -y centos-release-scl epel-release git libatomic libicu rsync
 ```
 
-### Build tools
+### Python 3
 
-Make sure `cmake`/`ctest` binaries are at least version 3. On CentOS, one way to achieve this is to symlink them into `/usr/local/bin`.
+Python 3.7 or higher is required.
+The following example installs Python 3.8.
 
 ```sh
+sudo yum -y install rh-python38
+# You may also want to add the following line to your .bashrc or equivalent.
+source /opt/rh/rh-python38/enable
+```
+
+### CMake 3
+
+[CMake][cmake] 3.17.3 or higher is required.
+The package manager has that, but we still need to link the name `cmake` to `cmake3`.
+Do similarly for `ctest`.
+
+```sh
+sudo yum install -y cmake3
 sudo ln -s /usr/bin/cmake3 /usr/local/bin/cmake
 sudo ln -s /usr/bin/ctest3 /usr/local/bin/ctest
 ```
 
-You could also symlink them into another directory that is on your `PATH`.
+[cmake]: https://cmake.org
 
-{{< note title="Note" >}}
+### /opt/yb-build
 
-We also use [Linuxbrew](https://github.com/linuxbrew/brew) to provide some of the third-party dependencies on CentOS.
-Linuxbrew allows us to create a portable package that contains its own copy of glibc and can be installed on most Linux distributions.
-However, we are transitioning away from using Linuxbrew and towards native toolchains on various platforms.
+By default, when running build, thirdparty libraries are not built, and pre-built libraries are downloaded.
+We also use [Linuxbrew][linuxbrew] to provide some of the third-party dependencies on CentOS.
+The build scripts automatically install these in directories under `/opt/yb-build`.
+In order for the build script to write under those directories, it needs proper permissions.
+One way to do that is as follows:
 
-Our build scripts may automatically install Linuxbrew in a directory such as `/opt/yb-build/brew/linuxbrew-<version>`.
-There is no need to add any of those directories to `PATH`.
+```sh
+sudo mkdir /opt/yb-build
+sudo chown "$(whoami)" /opt/yb-build
+```
 
-{{< /note >}}
+Alternatively, the build options `--no-download-thirdparty` and/or `--no-linuxbrew` can be specified.
+However, those cases may require additional, undocumented steps.
 
-### Java
+[linuxbrew]: https://github.com/linuxbrew/brew
 
-{{% readfile "includes/java.md" %}}
+### Ninja (optional)
+
+It is recommended to use [Ninja][ninja] for faster build.
```

Comment:
Most places, you can eliminate phrases like "we recommend," "it is recommended" (kill the passive voice!!!), and so on. It's pretty obvious we're recommending you do this, since it's in the instructions we're giving you. :)
```suggestion
Use [Ninja][ninja] for faster builds.
```